platform/kube/controller.go

```diff
@@ -399,8 +401,16 @@ func (c *Controller) GetIstioServiceAccounts(hostname string, ports []string) []
 // AppendServiceHandler implements a service catalog operation
 func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) error {
 	c.services.handler.Append(func(obj interface{}, event model.Event) error {
-		if svc := convertService(*obj.(*v1.Service), c.domainSuffix); svc != nil {
-			f(svc, event)
+		svc := *obj.(*v1.Service)
+
+		if svc.Namespace == meta_v1.NamespaceSystem {
```

Comment:
If we ever need to apply routing rules to kube api server, we will re-visit this. For now, there is no such requirement.